python/src/trezorlib/cardano.py

```diff
@@ -315,6 +321,69 @@ def create_withdrawal(withdrawal) -> messages.CardanoTxWithdrawalType:
     )
 
 
+def parse_auxiliary_data(auxiliary_data) -> messages.CardanoTxAuxiliaryDataType:
+    if auxiliary_data is None:
+        return None
+
+    AUXILIARY_DATA_MISSING_FIELDS_ERROR = ValueError(
+        "Auxiliary data is missing some fields"
+    )
+
+    if "type" not in auxiliary_data:
+        raise AUXILIARY_DATA_MISSING_FIELDS_ERROR
+
+    auxiliary_data_type = auxiliary_data["type"]
+    if auxiliary_data_type == messages.CardanoAuxiliaryDataType.BLOB:
+        if "blob" not in auxiliary_data:
+            raise AUXILIARY_DATA_MISSING_FIELDS_ERROR
+
+        return messages.CardanoTxAuxiliaryDataType(
+            type=auxiliary_data_type,
+            blob=bytes.fromhex(auxiliary_data["blob"]),
+        )
+    elif auxiliary_data_type == messages.CardanoAuxiliaryDataType.TUPLE:
+        if "metadata" not in auxiliary_data:
+            raise AUXILIARY_DATA_MISSING_FIELDS_ERROR
+
+        metadata = auxiliary_data["metadata"]
+        if "type" not in metadata:
+            raise AUXILIARY_DATA_MISSING_FIELDS_ERROR
+
+        metadata_type = metadata["type"]
+        if metadata_type == messages.CardanoMetadataType.CATALYST_REGISTRATION:
+            catalyst_registration = metadata["catalyst_registration_parameters"]
+            if not all(
+                k in catalyst_registration
+                for k in REQUIRED_FIELDS_CATALYST_REGISTRATION
+            ):
+                raise AUXILIARY_DATA_MISSING_FIELDS_ERROR
+
+            catalyst_registration_parameters = (
+                messages.CardanoCatalystRegistrationParametersType(
+                    voting_public_key=bytes.fromhex(
+                        catalyst_registration["voting_public_key"]
+                    ),
+                    staking_path=tools.parse_path(
+                        catalyst_registration["staking_path"]
+                    ),
+                    nonce=catalyst_registration["nonce"],
```

Comment:
Nice, yes it can. The same goes `validity_interval_start`, token `amount`, pool margin `numberator` and other `int`s which are unbounded in the code. Protobuf can handle it, python too but then it crashes on CBOR serialization. I am not sure what the best way to handle this is. Trezor Connect silently overflows.

I could wrap all the trezorlib int parsing into something similar to:
```
def parse_uint64(json_obj: dict, field_name: str) -> Optional[int]:
    uint64 = json_obj.get(field_name)
    if uint64 is None:
        return None

    if uint64 > 18446744073709551615:
        raise ValueError("%s is too large" % field_name)

    return uint64
```

I don't think this should be validated on the Cardano FW app level.

Reply:
I will raise this with Trezor guys.